### ISSUE TITLE
gradient of p2 norm does not agree with analytic solution

### ISSUE THREAD

**Issue by MasterXiong** [pytorch/pytorch#113527]
### 🐛 Describe the bug

I want to scale a vector by its p2 norm, i.e., `y = x / norm(x, p=2)`. The analytic gradient of this operation w.r.t. each element in the vector is `1 / ||x||_2 - x_i^2 / ||x||_2^3`. However, I found that the gradient computed by pytorch is different from this analytic solution. Not sure if I made some mistakes here or pytorch does not compute it correctly? 
Here is a minimal example to reproduce this inconsistent issue: 
```python
import torch
import torch.nn as nn

# Function to compute the gradient of vector normalization
def gradient_vector_norm(x):
    # Convert input to PyTorch tensor if not already
    x = torch.tensor(x, requires_grad=True, dtype=torch.float)

    # Compute the normalized vector
    norm_x = x.norm(p=2)
    y = torch.div(x, norm_x).sum()
    y.backward()

    return x.grad, norm_x

# Test the function
vector = [3.0, 4.0]
gradient, norm_x = gradient_vector_norm(vector)

print("Gradient w.r.t. each element:", gradient)
analytic_gradient = 1. / norm_x - torch.tensor(vector) ** 2 / (norm_x ** 3)
print (analytic_gradient)
```
Pytorch give a gradient of [ 0.0320, -0.0240], while the analytic solution is [0.1280, 0.0720], which are inconsistent.

### Versions

Collecting environment information...
PyTorch version: 2.0.0+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 ��ͥ���İ�
GCC version: (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0

Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.8.3 (default, Jul  2 2020, 17:30:36) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19041-SP0
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: GeForce MX250
Nvidia driver version: 425.25
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture=9

CurrentClockSpeed=1800

DeviceID=CPU0

Family=205

L2CacheSize=1024

L2CacheSpeed=

Manufacturer=GenuineIntel

MaxClockSpeed=1800

Name=Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz

ProcessorType=3

Revision=

Versions of relevant libraries:
[pip3] flake8==3.8.3
[pip3] numpy==1.19.5
[pip3] numpydoc==1.1.0
[pip3] torch==2.0.0
[pip3] torchfile==0.1.0
[pip3] torchtext==0.6.0
[conda] blas                      1.0                         mkl  
[conda] mkl                       2020.1                      216  
[conda] mkl-service               2.3.0            py38hb782905_0  
[conda] mkl_fft                   1.1.0            py38h45dec08_0  
[conda] mkl_random                1.1.1            py38h47e9c7a_0  
[conda] numpy                     1.18.5                   pypi_0    pypi
[conda] numpydoc                  1.1.0                      py_0  
[conda] torch                     2.0.0                    pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchtext                 0.6.0                    pypi_0    pypi
